CDP4JsonFileDal/JsonFileDal.cs

```diff
@@ -288,10 +314,22 @@ public override Task<IEnumerable<Thing>> Write(IEnumerable<OperationContainer> o
                 }
 
                 Logger.Info("Successfully exported the open session {1} to {0}.", path, this.Session.Credentials.Uri);
+
+                if (allErrors.Any())
+                {
+                    throw new ModelWarningException();
+                }
+            }
+            catch (ModelWarningException ex)
+            {
+                Logger.Warn("Not all things were exported to Annex.C3 file {0}.\n {1} Errors detected:\n {2}", path, allErrors.Count, $"{string.Join("\n", allErrors)}\n{ex.Message}");
+
+                throw new ModelWarningException($"Not all things were exported to Annex.C3 file {path}.\n {allErrors.Count} errors detected:\n {string.Join("\n", allErrors)}\n{ex.Message}");

```

Comment:
do we need to create a new instance of an exception, can we not just re throw the caught exception?

Reply:
We need to rethrow as exception data can come from multiple routes in the code